js/models/messages.js

```diff
@@ -1900,33 +1912,63 @@
       return message;
     },
 
-    async handleDataMessage(initialMessage, confirm) {
```

Comment:
`handleDataMessage` was split in several new methods:
* `handleGroupMessage()`
* `handleSessionRequest()`
* `shouldIgnoreBlockedGroup()`

also, a part of logic before in `message_receiver.js` is now here (session request logic for example)